### PR TITLE
add non blocking pipeline creation flow

### PR DESCRIPTION
**Pull request by karthikjeeyar**
**Fixes**: 
https://issues.redhat.com/browse/ODC-5048

**Analysis / Root cause**: 
As a user, I'd like to still be able to finish my add flow even if the Pipeline Run cannot be created for the added Pipeline.

**Solution Description**: 
Removed the pipelinerun creation request from the promise collection and added it in a try/catch block seperately.

**Screen shots / Gifs for design review**: 

**Before**:
Blocks the add flow redirection and shows the error.
image

**After**:
Does not block the add flow redirection and suppresses the pipelinerun creation error
non-blocking-flow
**Topology Sidebar**
image

**Unit test coverage report**:
import-submit-utils.spec.ts 
image
PipelineOverview.spec.tsx
image
pipeline-overview-utils.spec.ts
image

**Test setup:**
1. Install Openshift pipeilnes operator with `preview` channel to get 1.2 operator
2. Go to add from git flow
3. Use dotnet image `https://github.com/redhat-developer/s2i-dotnetcore-ex.git` and choose dotnet builder image
4. Add Pipeline option
5. click on create button

**Browser conformance**: 

- [x] Chrome
- [x] Firefox
- [ ] Safari
- [ ] Edge

cc: @andrewballantyne @bgliwa01  @openshift/team-devconsole-ux

### CODE REVIEW

**Comment by andrewballantyne** [comment]
Ah, I missed AC in the Story, apologizes for that @karthikjeeyar - If you look at the [design](https://docs.google.com/document/d/1dH8w11ZPKplScMwA93rQ0je8NPrTToKIikFtc7sZvBc/edit#heading=h.4me1hiepdkww), we have a need to show a message in the sidebar.

🤔 To do this, perhaps we find a way to add some form of `sessionStorage` (NOT `localStorage`) value when it fails?

> The read-only `sessionStorage` property accesses a session Storage object for the current origin. `sessionStorage` is similar to `localStorage`; the difference is that while data in `localStorage` doesn't expire, data in `sessionStorage` is cleared when the page session ends. 

Ref: [MDN Docs](https://developer.mozilla.org/en-US/docs/Web/API/Window/sessionStorage)

Our goal here is to just inform the user immediately, not to persist it indefinitely.

**Review by andrewballantyne — CHANGES_REQUESTED** [review]

File: frontend/packages/dev-console/src/components/import/__tests__/import-submit-utils.spec.ts
Comment (karthikjeeyar) [reply frontend/packages/dev-console/src/components/import/__tests__/import-submit-utils.spec.ts:244]: Pipeline resource creation call is not pushed into the requests array (uses await) , so i will add a spy on the creation call and expect it to not throw the error.

**Comment by karthikjeeyar** [comment]
@andrewballantyne I have updated the code to show the alert in the sidebar, updated screenshots and added more tests.

File: frontend/packages/dev-console/src/components/import/__tests__/import-submit-utils.spec.ts
Comment (andrewballantyne) [reply frontend/packages/dev-console/src/components/import/__tests__/import-submit-utils.spec.ts:249]: Ah so it does... apologizes for not looking at the code! I misremembered that the PipelineRun was added to the list and you removed it to use await with this PR. 👍

File: frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-overview/PipelineOverview.tsx
Comment (andrewballantyne) [inline frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-overview/PipelineOverview.tsx:40]: I think we want to just live read it on each render... without the hooks. This way wouldn't need to worry so much about unsetting it when the user clicks the 'x'.

```suggestion
  const showAlert = isPipelineNotStarted(name);
```

File: frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-overview/PipelineOverview.tsx
Comment (andrewballantyne) [inline frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-overview/PipelineOverview.tsx:45]:
```suggestion
        {(showAlert && pipelineRuns.length === 0) && (
```
If they start up a Pipeline Run we don't want to show it.

File: frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-overview/pipeline-overview-utils.ts
Comment (andrewballantyne) [inline frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-overview/pipeline-overview-utils.ts:22]: We want to hold onto the namespace as well... if you just do name, I could switch between projects and get unexpected results.

Plus you probably want to only write it back out when you modify it... no need to write the same value you got back to the storage if it didn't change.

```suggestion
export const setPipelineNotStarted = (namespace: string, pipelineName: string): void => {
  if (!pipelineName || !namespace) return;

  const pipelineData = getNotStartedPipelines();
  if (!pipelineData[namespace]) {
    pipelineData[namespace] = [];
  }
  if (!pipelineData[namespace].includes(pipelineName)) {
    pipelineData[namespace].push(pipelineName);
    sessionStorage.setItem(PIPELINE_RUN_AUTO_START_FAILED, JSON.stringify(pipelineData));
  }
};
```

File: frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-overview/pipeline-overview-utils.ts
Comment (andrewballantyne) [inline frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-overview/pipeline-overview-utils.ts:4]: Based on my last comment... needs to be an object

File: frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-overview/pipeline-overview-utils.ts
Comment (andrewballantyne) [inline frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-overview/pipeline-overview-utils.ts:11]: Based on my last(last) comment... need namespace here.

**Review by andrewballantyne — CHANGES_REQUESTED** [review]

File: frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-overview/PipelineOverview.tsx
Comment (karthikjeeyar) [reply frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-overview/PipelineOverview.tsx]: removed UseEffect but still using local state as the re render is not quick enough to remove the Alert.

File: frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-overview/pipeline-overview-utils.ts
Comment (karthikjeeyar) [reply frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-overview/pipeline-overview-utils.ts]: @andrewballantyne Updated the code to store/remove a namespaced session object.

File: frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-overview/PipelineOverview.tsx
Comment (andrewballantyne) [inline frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-overview/PipelineOverview.tsx:36]: This is not reactive 😞  

I'll need to do some investigation into why we can't just use it in the render.

File: frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-overview/PipelineOverview.tsx
Comment (andrewballantyne) [reply frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-overview/PipelineOverview.tsx]: Ah I see... my bad. 🤔

File: frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-overview/PipelineOverview.tsx
Comment (andrewballantyne) [reply frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-overview/PipelineOverview.tsx:36]: Ah okay, I see what is happening here... yeah your original solution made more sense. I'm not sure if we can entirely base our data on the sessionStorage 🤔  (which was my original reason for my suggestion)

Comment (andrewballantyne) [reply frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-overview/PipelineOverview.tsx:36]: Yeah, I think we can't get a "single state" of storage here... which is unfortunately. Okay, reset it back to the useEffect hook and we should be good to go.

```
  const [showAlert, setShowAlert] = React.useState(isPipelineNotStarted(name, namespace));

  React.useEffect(() => {
    setShowAlert(isPipelineNotStarted(name, namespace));
  }, [name, namespace]);
```

Sorry for this round-about solution... you had it right the first time.

**Review by andrewballantyne — APPROVED** [review]
Just need to rebase and reimplement the hook @karthikjeeyar - Get a LGTM from someone on the team or if I see the push I'll give it. Thanks!

**Comment by karthikjeeyar** [comment]
Rebased and added the hook back.

**Review by andrewballantyne — APPROVED** [review]
/lgtm

**Pull request merged**
